meson.build

```diff
@@ -0,0 +1,41 @@
+project(
+    'erfa', 'c',
+    version: '2.0.0',
+    meson_version: '>=0.37.0',
+)
+
+cc = meson.get_compiler('c')
+confdata = configuration_data()
+
+# The historic versions use libtool-compatible versioning.
+# This uses some gnarly math to define ABI versions, which we replicate here.
+# The general formula is:
+#   libtool: C:R:A
+#   -soname: (C - A).A.R
+libtool_version = [9, 0, 8]
+soversion = '@0@.@1@.@2@'.format(
+    libtool_version[0] - libtool_version[2],
+    libtool_version[2],
+    libtool_version[1],
+)
+
+# API versions
+version = meson.project_version().split('.')
+confdata.set_quoted('PACKAGE_VERSION', meson.project_version())
+confdata.set('PACKAGE_VERSION_MAJOR', version[0])
+confdata.set('PACKAGE_VERSION_MINOR', version[1])
+confdata.set('PACKAGE_VERSION_MICRO', version[2])
+
+confdata.set_quoted('SOFA_VERSION', '20210512')
+
+
+libm = cc.find_library('m', required: false)
```

Comment:
Why not required? (Clearly do not understand yet how this works, so mostly out of curiosity!)

Reply:
For Windows support! :D

On Windows, there's no libm, and can't be. Whatever Windows support you have (I think there are math symbols already available? but I'm no Windows expert) will work without libm.

Libm is guaranteed to exist on POSIX platforms in some form, and to come with libc itself. In some cases it may actually be part of libc, in which case there may not be a file for it, but the compiler is required to accept an `-lm` argument which it may resolve to an empty static archive or simply delete from the command line, as it pleases -- POSIX doesn't mandate an implementation.

So, basically:
- libm may or may not exist as a library to look up
- if it is needed on that platform, it will always be there
- if it's not found, that is because it won't be needed

Reply:
Ah, thanks!

Reply:
CI results for Windows are e.g. here: https://github.com/eli-schwartz/erfa/actions/runs/3148215060/jobs/5118511545

```
C compiler for the host machine: cl (msvc 19.33.31630 "Microsoft (R) C/C++ Optimizing Compiler Version 19.33.31630 for x64")
C linker for the host machine: link link 14.33.31630.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Library m found: NO
```

But in the end it fully compiles and runs the selfcheck executables successfully, without an `m` library. The functions needed by erfa are part of the CRT.